### ISSUE TITLE
terra e:enable: Unable to clone repository. Check app settings and try again.

### ISSUE THREAD

**Issue by jlyon** [terra-ops/terra-cli#79]
I have been trying to enable an app with a large repo (1gb) and I am getting this error:

```
Unable to clone repository. Check app settings and try again.
```

Oddly enough, it looks like the repo is getting cloned properly (and completely).  If I run `terra e:enable` again the env gets enabled and all is well.

**Comment by jonpugh** [comment]
might be timing out?

**Comment by jlyon** [comment]
yeah, i think maybe the php script is timing out or something? the full repo does get cloned. I've had similar issues with other smaller repos on my localhost.

**Comment by jonpugh** [comment]
Hmm... Since we use https://github.com/cpliakas/git-wrapper it is really in their code...

We call "clone" at https://github.com/terra-ops/terra-cli/blob/master/src/terra/Factory/EnvironmentFactory.php#L85

That error is from our code and is really dumb... we need better exceptions thrown there.

Let's dig in and figure out what is really causing the problem.

Can you post additional error messages right before that one?

We can also look into if the git-wrapper uses symfony Process and setTimeout(null).

**Comment by asgorobets** [comment]
GitWrapper has a timeout of 60 seconds if not specified otherwise.
We might want to modify this timeout for terra, as GitWrapper is configurable to do that.

For example:

```
$wrapper = new GitWrapper();
$wrapper->setTimeout($this->getTimeout());
```

**Comment by kynetiv** [comment]
Just wanted to share I ran into this where cloning was taking too long for whatever reason. I thought I might just clone it myself somewhere and let `environment:add` know the path to the cloned copy but then ran into the reverse error that:

`Git clone already exists at that path, but it is not for this app.`

As a workaround I ended up just manually adding the environment to the ~/.terra/terra file and it was fine.
